src/core/EdFi.DataManagementService.Core/Middleware/DisallowDuplicateReferencesMiddleware.cs

```diff
@@ -130,4 +158,30 @@ private static string ExtractArrayName(string path)
         string[] parts = path.Split('.');
         return parts.Length > 1 ? parts[1].Trim('[', ']', '*') : string.Empty;
     }
+
+    private static Dictionary<string, Dictionary<int, List<DescriptorReference>>> GroupByArrayNameAndIndex(
+        IList<DescriptorReference> jsonNodes
+    )
+    {
+        var regex = ArrayNameRegex();
```

Comment:
I'd either drop this and inline `ArrayNameRegex()` or rename to `var arrayNameRegex = ArrayNameRegex() `